nixos/modules/services/misc/gitea.nix

```diff
@@ -162,6 +162,44 @@ in
             <manvolnum>7</manvolnum></citerefentry>.
           '';
         };
+
+        backupDir = mkOption {
+          type = types.str;
+          default = "${cfg.stateDir}/dump";
+          description = "Path to the dump files.";
+        };
+      };
+
+      ssh = {
```

Comment:
I still feel that these options provide no extra value. All these options do is copy values to the configuration file which can be accomplished with this configuration instead:
```
services.gitea.settings.DISABLE_SSH = false;
services.gitea.settings.SSH_PORT = 22;

# instead of

services.gitea.ssh.enable = true;
services.gitea.ssh.clonePort = 22;
```

You see the difference between the `ssh` and `lfs` options you provided: the `lfs` options "do" more in the module. They don't just copy values into configuration, they tweak the `systemd` service (`ReadWritePaths`), they provision directories (`systemd.tmpfiles.rules`).

Reply:
In my case, ssh didn't work for me right away, I had to look for the reason. The `clonePort` description provides a quick fix if NAT is in use.

Reply:
How does `NAT` changes anything about the way the port is used? Is port forwarding not required anyway?

Reply:
I have DNAT rule written on my router. Forwards from port 2222 to port 22. Don't need to change anything on the server.

Reply:
Ok. I improved the explanation: https://github.com/NixOS/nixpkgs/pull/94291/files#r465550727
because this is due to port forwarding and not NAT. Even with NAT the external port and internal port could match.